python/llm/portable-zip/chat.py

```diff
@@ -276,31 +321,43 @@ def auto_select_model(model_name):
     parser = argparse.ArgumentParser()
     parser.add_argument("--model-path", type=str, help="path to an llm")
     parser.add_argument("--start-size", type=int, default=4, help="start_size of kv_cahce")
+    parser.add_argument("--recent-size", type=int, default=2000)
     args = parser.parse_args()
 
     model_path = args.model_path
     start_size = args.start_size
+    recent_size = args.recent_size
 
     model = auto_select_model(model_path)
     model = optimize_model(model)
 
     tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True)
+    print("model.config.architectures[0] is: ", model.config.architectures[0])
+    print("model.config.architectures[0] == 'BaiChuanForCausalLM' is: ", model.config.architectures[0] == "BaiChuanForCausalLM")
+    print("model.config.vocab_size == 64000 is: ", model.config.vocab_size == 64000)
 
     if model.config.architectures is not None and model.config.architectures[0] == "QWenLMHeadModel":
         stop_words = get_stop_words_ids("Qwen", tokenizer=tokenizer)
-        kv_cache = StartRecentKVCache(start_size=start_size, k_seq_dim=1, v_seq_dim=1)
+        kv_cache = StartRecentKVCache(start_size=start_size,
+                                      k_seq_dim=1,
+                                      v_seq_dim=1,
+                                      recent_size=recent_size)
         qwen_stream_chat(model=model, tokenizer=tokenizer,kv_cache=kv_cache, stop_words=stop_words)
     elif model.config.architectures is not None and model.config.architectures[0] == "ChatGLMModel":
         chatglm3_stream_chat(model=model, tokenizer=tokenizer)
     elif model.config.architectures is not None and model.config.architectures[0] == "LlamaForCausalLM":
-        kv_cache = StartRecentKVCache(start_size=start_size)
+        kv_cache = StartRecentKVCache(start_size=start_size, recent_size=recent_size)
         if "yi" in model_path.lower():
             stop_words = get_stop_words_ids("Yi", tokenizer=tokenizer)
             yi_stream_chat(model=model, tokenizer=tokenizer, kv_cache=kv_cache, stop_words=stop_words)
         else:
             llama_stream_chat(model=model, tokenizer=tokenizer, kv_cache=kv_cache)
+    elif model.config.architectures[0] == "BaichuanForCausalLM" and model.config.vocab_size == 64000:
+        # Baichuan-13B-Chat
```

Comment:
Only one concern, by using the previous chat history mode, the 1st token latency will by much longer after several turns.

Reply:
Yes, but Baichuan-13B cannot use our past_key_values method.

Reply:
Other models will use our kv_cache method